lightning/restore/restore.go

```diff
@@ -343,6 +351,47 @@ func (rc *RestoreController) restoreSchema(ctx context.Context) error {
 	return nil
 }
 
+// verifyCheckpoint check whether previous task checkpoint is compatible with task config
+func verifyCheckpoint(cfg *config.Config, taskCp *TaskCheckpoint) error {
+	if taskCp == nil {
+		return nil
+	}
+	// always check the backend value even with 'check-requirements = false'
+	retryUsage := "destroy all checkpoints and remove all restored tables and try again"
+	if cfg.TikvImporter.Backend != taskCp.Backend {
+		return errors.Errorf("config 'tikv-importer.backend' value '%s' different from checkpoint value '%s', please %s", cfg.TikvImporter.Backend, taskCp.Backend, retryUsage)
+	}
+
+	if cfg.App.CheckRequirements {
+		errorFmt := "config '%s' value '%s' different from checkpoint value '%s'. You may set 'check-requirements = false' to skip this check or " + retryUsage
+		if cfg.Mydumper.SourceDir != taskCp.SourceDir {
+			return errors.Errorf(errorFmt, "mydumper.data-source-dir", cfg.Mydumper.SourceDir, taskCp.SourceDir)
+		}
+
+		if cfg.TikvImporter.Backend == config.BackendLocal && cfg.TikvImporter.SortedKVDir != taskCp.SortedKVDir {
+			return errors.Errorf(errorFmt, "mydumper.sorted-kv-dir", cfg.TikvImporter.SortedKVDir, taskCp.SortedKVDir)
+		}
+
+		if cfg.TikvImporter.Backend == config.BackendImporter && cfg.TikvImporter.Addr != taskCp.ImporterAddr {
+			return errors.Errorf(errorFmt, "tikv-importer.addr", cfg.TikvImporter.Backend, taskCp.Backend)
+		}
+
+		if cfg.TiDB.Host != taskCp.TiDBHost {
+			return errors.Errorf(errorFmt, "tidb.host", cfg.TiDB.Host, taskCp.TiDBHost)
+		}
+
+		if cfg.TiDB.Port != taskCp.TiDBPort {
+			return errors.Errorf(errorFmt, "tidb.port", cfg.TiDB.Port, taskCp.TiDBPort)
+		}
+
+		if cfg.TiDB.PdAddr != taskCp.PdAddr {
```

Comment:
`PdAddr` can be nil, maybe let it pass when `cfg.TiDB.PdAddr` or `taskCp.PdAddr` is nil.

Reply:
I think there is no reasonable reason for user to change pdAddr to empty when resume from checkpoint? And since `cfg.TiDB.PdAddr` maybe impact the target cluster addr, so I think this check is necessary, and user can still skip this check by set `check-requirements = false`